internal/store/bundle.go

```diff
@@ -35,24 +35,38 @@ type bundleStore struct {
 	path string
 }
 
-func (b *bundleStore) Store(ref reference.Named, bndle *bundle.Bundle) error {
-	path, err := b.storePath(ref)
+func (b *bundleStore) Store(ref reference.Named, bndle *bundle.Bundle) (reference.Reference, error) {
+	digest, err := b.storeByDigest(ref, bndle)
 	if err != nil {
-		return errors.Wrapf(err, "failed to store bundle %q", ref)
+		return nil, err
 	}
-	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
-		return errors.Wrapf(err, "failed to store bundle %q", ref)
+
+	if ref != nil {
+		dir, err := b.storePath(ref)
```

Comment:
nit: Can we extract this to a `storeByRef` function? Just to get some symmetry here 😸